lib/iex/lib/iex/helpers.ex

```diff
@@ -457,13 +474,29 @@ defmodule IEx.Helpers do
       13
       iex(2)> value
       13
+      iex(3)> import_file "nonexisting.file.ex", optional: true
+      nil
+
   """
-  defmacro import_file(path) when is_binary(path) do
+  defmacro import_file(path, opts \\ [])
+
+  defmacro import_file(path, opts) when is_binary(path) do
+    optional? = Keyword.get(opts, :optional, false)
     path = Path.expand(path)
-    Code.string_to_quoted! File.read!(path), file: path
+
+    case File.read(path) do
+      {:ok, contents} ->
+        Code.string_to_quoted!(contents, file: path)
+      {:error, reason} ->
+        if optional? do
+          nil
+        else
+          raise(File.Error, action: "read file", path: path, reason: reason)
```

Comment:
It could fail for many reasons... we are only worried if it exists or not. You can do something like this:

    if not optional? || File.exists?(...) do
      File.read! Code.string_to_quoted!
    end

Wdyt?